specification/indices/put_data_stream_settings/IndicesPutDataStreamSettingsResponse.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexSettings } from '@indices/_types/IndexSettings'
+import { IndexName } from '@_types/common'
+
+export class Response {
+  /** @codegen_name data_stream_settings_results */
+  body: {
+    data_streams: Array<DataStreamSettings>
+  }
+}
+
+export class DataStreamSettings {
+  name: IndexName
```

Comment:
There should be a description for all properties.
```suggestion
  /** The data stream name. */
  name: IndexName
```